server/src/main/java/io/crate/execution/engine/aggregation/AggregationFunction.java

```diff
@@ -114,8 +121,27 @@ public TPartial removeFromAggregatedState(RamAccounting ramAccounting,
                                                 "not removable cumulative");
     }
 
+    @VisibleForTesting
     @Nullable
     public DocValueAggregator<?> getDocValueAggregator(List<DataType<?>> argumentTypes, List<MappedFieldType> fieldTypes, List<Literal<?>> optionalParams) {
         return null;
     }
+
+    @Nullable
+    public DocValueAggregator<?> getDocValueAggregator(List<Symbol> aggregationReferences,
```

Comment:
Do you think it would make sense to instead change the existing `getDocValueAggregator` overload? Otherwise there's the question of which one to use/implement, and it looks like this one is strictly more powerful then the other - so why not just have one?

Reply:
Thank you for review, @mfussenegger. I am keeping the more powerful version and adjusted test cases accordingly.